File: Easy/High Effort vs Low Effort/high-effort-vs-low-effort.cpp
class Solution {
  public:
  
  int n;
  vector<vector<int>> dp;
  
    int maxTask(vector<int>& h, vector<int>& l) {
        // code here
        n = h.size();
        dp.resize(n , vector<int>(2 , -1));
        
        return solve(0 , 0 , h , l);
        //0 for false and 1 for true
        
    }
    
    int solve(int index , int prevtask ,vector<int>& h, vector<int>& l){
        if (index == n){
            return 0;
        }
        
        if (dp[index][prevtask] != -1) return dp[index][prevtask];
        
        // Eat 5 Start
        int nothing = solve(index+1 , 0 , h , l);
        
        int low = l[index] + solve(index+1 , 1 , h , l);
        
        int high = 0;
        if (prevtask == 0){
            high = h[index] + solve(index+1 , 1 , h , l);
        }
        
        return dp[index][prevtask] = max({nothing , high , low});
        
    }
    
};

File: Easy/High Effort vs Low Effort/high-effort-vs-low-effort.py
class Solution:
    def maxTask(self, h: list[int], l: list[int]) -> int:
        # code here
        n = len(h)
        dp = [[-1 for _ in range(2)] for _ in range(n)]
        
        def solve(index , prevtask):
            if index == n:
                return 0
                
            state = 1 if prevtask else 0
            
            if dp[index][state] != -1: return dp[index][state]
            
            # Eat 5 star
            nothing = solve(index+1 , False)
            
            # Do low effort
            low = l[index] + solve(index+1 , True)
            
            #Do full high effort
            
            high = 0
            if not prevtask:
                high = h[index] + solve(index+1 , True)
            
            dp[index][state] = max([nothing , low , high])
            
            return dp[index][state]
        
        return solve(0 , False)
            
            